MMDP-334 Enable search and filtering of stakeholders

fgkinus opened the pull request:
**What does this PR do?**

- add functionality for searching under the Stakeholders Directory section
- add functionality for filtering search results by State or Local Government Area
- add modal for viewing stakeholder details

**Description of Task to be completed?**

As an administrator, I want to be able to search stakeholders displayed in the 'Stakeholders Directory' section of the 'Resources' page by entering search criteria in a search box.I should also be able to filter the selection by state by applying filters.

**How should this be manually tested?**

```
git clone https://github.com/andela/mmdp_site
cd mmdp_site
git checkout ft-search-filter-stakeholders-MMDP-334
yarn run: server
yarn client:dev
```
- browse to `<localhost>:<port>/resources/stakeholders`

**Any background context you want to provide?**

To add stakeholders, reference the pull request for adding new stakeholders : 
https://github.com/andela/mmdp_site/pull/47

Furthermore, this Implementation requires mongo transactions to be enabled. Mongoose requires mongo version >4.0.0 and replica sets to be enabled. In line with this please refer to this guide for setting up the development environment and also more about replica sets.

about mongo transactions: https://mongoosejs.com/docs/transactions.html
setup-guide : http://thecodebarbarian.com/introducing-run-rs-zero-config-mongodb-runner.html


**What are the relevant Jira issues?**

[MMDP-334](https://viisaus.atlassian.net/browse/MMDP-334)

**Screenshots (if appropriate)**
- list stakeholders
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/20369585/53785282-01ec8e80-3f29-11e9-86e5-7497cd33466a.png">

- modal for viewing stakeholder details
![image](https://user-images.githubusercontent.com/20369585/53785671-63612d00-3f2a-11e9-9186-63ebae5aafca.png)

- results filtered by state
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/20369585/53785309-19c41280-3f29-11e9-845f-2e5103012f6d.png">

- search by string
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/20369585/53785333-2e080f80-3f29-11e9-9248-152b114b043a.png">

- filter by LGA
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/20369585/53785381-57c13680-3f29-11e9-89c4-1bb5cda31ed9.png">

p8ul commented:
Good work @fgkinus. Do you mind fixing the linting issues.
<img width="973" alt="screenshot 2019-03-08 at 09 36 39" src="https://user-images.githubusercontent.com/17355400/54012312-bb946b00-4186-11e9-8dfb-a9d07e3a5013.png">

p8ul commented on server/routes/api/stakeholdersDirectory.js:
```diff
@@ -118,6 +119,7 @@ export const update = async (req, res) => {
 
 export const list = async (req, res) => {
   try {
+    // const stakeholderDirectory =
```
Please, don't commit commented out code

fgkinus replied:
I have removed the commented code

fgkinus commented:
> Good work @fgkinus. Do you mind fixing the linting issues.
> <img alt="screenshot 2019-03-08 at 09 36 39" width="973" src="https://user-images.githubusercontent.com/17355400/54012312-bb946b00-4186-11e9-8dfb-a9d07e3a5013.png">

I have fixed this and removed the unused imports

fgkinus commented:
> Do you mind center aligning this loader.
> <img alt="screenshot 2019-03-08 at 10 15 21" width="1126" src="https://user-images.githubusercontent.com/17355400/54013720-411a1a00-418b-11e9-99fe-63820a4c0662.png">

I have fixed this and increased the width of the loader
<img width="1257" alt="image" src="https://user-images.githubusercontent.com/20369585/54015882-2eefaa00-4192-11e9-9bd7-9976ecdbf0f4.png">
